packages/kbn-unified-data-table/src/components/data_table.tsx

```diff
@@ -375,6 +375,13 @@ export interface UnifiedDataTableProps {
    * Optional extra props passed to the renderCellValue function/component.
    */
   cellContext?: EuiDataGridProps['cellContext'];
+
+  /**
+  Optional props passed to Columns to display provided labels as column names instead of field names.
+  This object maps column field names to their corresponding display labels.
+  These labels will take precedence over the data view field names.
+  */
+  columnHeaders?: Record<string, string>;
```

Comment:
nit: A more robust design would be to refactor `columns` to a list of objects instead of ids, to extend the information we hold on each column.